modules/apps/frontend-js/frontend-js-loader-modules-extender/src/main/java/com/liferay/frontend/js/loader/modules/extender/internal/servlet/JSResolveModulesServlet.java

```diff
@@ -134,9 +154,26 @@ private List<String> _getModuleNames(HttpServletRequest httpServletRequest)
 		return Collections.emptyList();
 	}
 
+	@Reference
+	private AbsolutePortalURLBuilderFactory _absolutePortalURLBuilderFactory;
+
 	@Reference
 	private BrowserModulesResolver _browserModulesResolver;
 
-	private volatile String _etag;
+	private final AtomicReference<State> _state = new AtomicReference<>();
+
+	private static class State {
+
+		public State() {
+			String hash = String.valueOf(UUID.randomUUID());
```

Comment:
If we get different uuids in different cluster nodes, and send the hash to the browser, we must guarantee that the same node will receive resolution requests. Otherwise, if incoming resolutions URLs are dispathed to different nodes, wouldn't we start a "redirection party"?

Reply:
Yes :scream:

Reply:
So, it really depends on the case whether this modification is better or worse than the previous situation...

Reply:
I'm going to investigate how costly it is to compute a real hash of the NPMRegistry state.

Reply:
I've pushed a new commit to fix this -> https://github.com/liferay-frontend/liferay-portal/pull/2883/commits/1586f249de48a4279176556488f4479427fb363d

The only caveat is that computing the hash takes a bit long (2.5s in my PC). This is something to add to the startup time of the server, though it shouldn't be horrible, since it is not blocking the other startup actions.